scanner/src/main/kotlin/Scanner.kt

```diff
@@ -83,6 +84,58 @@ class Scanner(
     private val archiver: FileArchiver? = scannerConfig.archive.createFileArchiver(),
     fileListStorage: ProvenanceFileStorage = scannerConfig.fileListStorage.createStorage()
 ) {
+    companion object {
+        /**
+         * Return a map of VCS paths for each [KnownProvenance] contained in [provenances].
+         */
+        fun getVcsPathsForProvenances(provenances: Set<ProvenanceResolutionResult>) =
```

Comment:
Can you explain for what purpose these are going to be re-used? 

(These are the internal of mapping `OrtResult`s` internal representation to `OrtResult`s external representation.
This is a bit complex bit, and I believe we should avoid making this any more complex. If it's just about re-use, maybe this is ok, but even that would make it harder to change things. If it's about extending functionality in order to re-use, then it will be imo more problematic.)

Reply:
See https://github.com/eclipse-apoapsis/ort-server/pull/1358#discussion_r1837705433.

Reply:
Can you please sahre the exact use case in ORT server?

Reply:
If the linked PR and commit message rationale does not provide enough context, I'll leave it to @wkl3nk to explain.

Reply:
The ORT scanner creates a perfect result, filtering out any non-relevant scan results. 
But actually ORT Server, when calling the ORT scanners, **does not use this (perfect) results returned**. I guess because they are already stored in the database cache of the scanner, and one did not want to store the results twice. But this creates the problem in ORT server, that every time the scan results need to be fetched from the database cache, they need to be filtered (for VCS path). And the logic to filter them should be exactly the same as in the ORT scanner. This is the reason for having this code reusable, instead to copy-and-paste it and risk that the logic drifts away over time.

Reply:
I see, so is it necessary to expose these functions individually, or would it suffice to expose a single function which does the job? (to minimize exposing internals)

Apart from that, I would prefer to be careful not to mix this internal mapping code with anything else, which is why I propose to put them not into `Scanner` but into a dedicated file.